airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ResetActivity.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.airbyte.config.StreamDescriptor;
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import java.util.List;
+import java.util.UUID;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@ActivityInterface
+public interface ResetActivity {
+
+  @Data
+  @NoArgsConstructor
+  @AllArgsConstructor
+  class ResetInput {
+
+    private UUID connectionId;
+
+  }
+
+  @Data
+  @NoArgsConstructor
+  @AllArgsConstructor
+  class GetResetsOutput {
+
+    private List<StreamDescriptor> streamDescriptorList;
+
+  }
+
+  /**
+   * Return the streams being reset for a given connection id
+   */
+  @ActivityMethod
+  GetResetsOutput getStreamResets(UUID connectionId);
+
+  /**
+   * Creates a stream_reset record for each stream descriptor passed in
+   */
+  @ActivityMethod
+  void createStreamResets(UUID connectionId, List<StreamDescriptor> streamDescriptorList);
+
+  /**
+   * Deletes the stream_reset record corresponding to each stream descriptor passed in
+   */
+  @ActivityMethod
+  void deleteStreamResets(UUID connectionId, List<StreamDescriptor> streamDescriptorList);
```

Comment:
I think we try to have every activity method implemented only take in a single argument, e.g. a `DeleteStreamResetsInput` object in this case, so that new "arguments" can be added to activity methods in a backwards-compatible way (since Temporal stores and replays activity inputs and outputs - as long as the signature of the activity methods does change, I think Temporal is able to "replay" the input and output of an activity that was ran on an old version)

Reply:
ah okay, I wasn't sure why some of the activities were constructed that way. will make that update.